### PR TITLE
global snapshot calculation minor issues

### PR DESCRIPTION
**Pull request by devyetii**
## Fixes
- `getblobbers` endpoint returns `saved_data` always zero.
## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

**Review by dabasov — APPROVED** [review]
lgtm

**Review by peterlimg — APPROVED** [review]
LGTM

**Comment by peterlimg** [comment]
@devyetii please check and fix failed system test https://github.com/0chain/0chain/actions/runs/4457387334/jobs/7828507331#step:6:2453

**Comment by devyetii** [comment]
I've added the do-not-merge flag as I'll be fixing all the minor issues in snapshot calculation here. Will be able to merge as soon as I finish adding system tests for global snapshots

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | 420-system-tests-for-global-historic-metrics-for-graphs  |
| gosdk  | staging  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | fix/getblobbers-return-saved_data  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

File: code/go/0chain.net/smartcontract/storagesc/allocation.go [hunk 1]
```diff
@@ -335,6 +336,16 @@ func (sc *StorageSmartContract) newAllocationRequestInternal(
 			zap.Error(err))
 		return "", common.NewError("allocation_creation_failed", err.Error())
 	}
+	i, _ := txn.Value.Int64()
+	balances.EmitEvent(event.TypeStats, event.TagLockWritePool, sa.ID, event.WritePoolLock{
+		Client:       txn.ClientID,
+		AllocationId: sa.ID,
+		Amount:       i,
+	})
+	transfer := state.NewTransfer(txn.ClientID, txn.ToClientID, txn.Value)
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/storagesc/allocation.go]: it's done inside `addToWritePool `

**Review by dabasov — APPROVED** [review]
lgtm

**Pull request merged**